components/ProfileElements/ContactInformationModal.jsx

```diff
@@ -0,0 +1,64 @@
+import { View } from "react-native";
+import GeneralModal from "../GeneralModal/GeneralModal.jsx";
+import { firebase_auth, firebase_db } from "../../firebase.config.js";
+import { useState, useEffect } from "react";
+import EditInfo from "../Form/EditInfo.jsx";
+``;
+import { set, ref, get } from "firebase/database";
+import LoadingWheel from "../Loading/LoadingWheel.jsx";
+
+
+export default function ContactInformationModal(props) {
+    const db = firebase_db;
+    const auth = firebase_auth;
+    const [loading, setLoading] = useState(true);
+    const [initialContactNo, setInitialContactNo] = useState("");
+    const [contactNo, setContactNo] = useState("");
+    const [initialEmail, setInitialEmail] = useState(auth.currentUser.email);
+    const [email, setEmail] = useState(auth.currentUser.email);
+    useEffect(() => {
+        const userRef = ref(db, "users/" + auth.currentUser.uid);
+        get(userRef).then((userSnapshot) => {
+            if (!userSnapshot.exists()) return;
+            const userData = userSnapshot.val();
+            setContactNo(userData.contactNo);
+            setInitialContactNo(userData.contactNo);
+            setLoading(false);
+        });
+    }, []);
+    return (
+        <GeneralModal closeModal={props.closeModal}>
+            {loading ? (
+                <LoadingWheel />
+            ) : (
+                <View className="flex flex-col items-center h-full">
+                    <EditInfo
+                        title="Contact Number"
+                        variableToEdit={contactNo}
+                        initialValue={initialContactNo}
+                        onChangeFunction={(text) => {
+                            setContactNo(text);
+                        }}
+                        onSubmitFunction={async() => {
```

Comment:
this is async with no await on the set, next is async with an await and in the other document there's a non-async func with a set needs to be consistent

Reply:
is this tested now that you got rid of the asyncs?